File: word-clock/ClockElement.cpp
#include "ClockElement.h"
#include "arduino.h"

ClockElement::ClockElement(){ }

ClockElement::ClockElement(int numericValueAM, int numericValuePM, int from, int to, CLOCK_ELEMENT_TYPE type){
  _numericValueAM = numericValueAM;
    _numericValuePM = numericValuePM;
  _from = from;
  _to = to;
  _type = type;
}

int ClockElement::GetRangeFrom(){
  return _from;  
}

int ClockElement::GetRangeTo(){
  return _to;  
}

int ClockElement::GetNumericValueAM() {
  return _numericValueAM;
}

int ClockElement::GetNumericValuePM() {
  return _numericValuePM;
}

int ClockElement::GetNumericValuesArrayLength() {
  return _to - _from;
}

CLOCK_ELEMENT_TYPE ClockElement::GetClockElementType(){
  return this->_type;  
}

File: word-clock/99crafts/ClockElement.py
from enum import Enum


class ClockElementType(Enum):
    HOUR = 1
    MINUTE = 2
    WORD = 3


class ClockElement:
    def __init__(self, numeric_value_am, numeric_value_pm, led_from, led_to, element_type):
        self.numeric_value_am = numeric_value_am
        self.numeric_value_pm = numeric_value_pm
        self.led_from = led_from
        self.led_to = led_to
        self.element_type = element_type

    def get_range_from(self):
        return self.led_from

    def get_range_to(self):
        return self.led_to

    def get_numeric_value_am(self):
        return self.numeric_value_am

    def get_numeric_value_pm(self):
        return self.numeric_value_pm